contracts/Committee.sol

```diff
@@ -0,0 +1,548 @@
+pragma solidity 0.5.16;
+
+import "./spec_interfaces/IContractRegistry.sol";
+import "./spec_interfaces/ICommittee.sol";
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "./spec_interfaces/IValidatorsRegistration.sol";
+
+/// @title Elections contract interface
+contract Committee is ICommittee, Ownable {
+	address[] topology;
+
+	struct Member {
+		bool member;
+		bool readyForCommittee;
+		uint256 readyToSyncTimestamp;
+		uint256 weight;
+	}
+	mapping (address => Member) members;
+
+	uint minimumWeight;
+	address minimumAddress;
+	uint minCommitteeSize;
+	uint maxCommitteeSize;
+	uint maxStandbys;
+	uint readyToSyncTimeout;
+
+	// Derived properties
+	uint committeeSize;
+	uint readyForCommitteeCount;
+	uint oldestReadyToSyncStandbyPos;
+
+	modifier onlyElectionsContract() {
+		require(msg.sender == contractRegistry.get("elections"), "caller is not the elections");
+
+		_;
+	}
+
+	constructor(uint _minCommitteeSize, uint _maxCommitteeSize, uint _minimumWeight, uint _maxStandbys, uint256 _readyToSyncTimeout) public {
+		require(_maxCommitteeSize > 0, "maxCommitteeSize must be larger than 0");
+		require(_maxStandbys > 0, "maxStandbys must be larger than 0");
+		require(_readyToSyncTimeout > 0, "readyToSyncTimeout must be larger than 0");
+		require(_maxStandbys > 0, "maxStandbys must be larger than 0");
+		minCommitteeSize = _minCommitteeSize;
+		maxCommitteeSize = _maxCommitteeSize;
+		minimumWeight = _minimumWeight;
+		maxStandbys = _maxStandbys;
+		readyToSyncTimeout = _readyToSyncTimeout;
+	}
+
+	/*
+	 * Methods restricted to other Orbs contracts
+	 */
+
+	/// @dev Called by: Elections contract
+	/// Notifies a weight change for sorting to a relevant committee member.
+    /// weight = 0 indicates removal of the member from the committee (for exmaple on unregister, voteUnready, voteOut)
+	function memberWeightChange(address addr, uint256 weight) external onlyElectionsContract returns (bool committeeChanged, bool standbysChanged) {
+		if (isMember(addr)) {
+			members[addr].weight = weight;
+			return _rankValidator(addr);
+		}
+		return (false, false);
+	}
+
+	function memberReadyToSync(address addr) external onlyElectionsContract returns (bool committeeChanged, bool standbysChanged) {
+		if (isMember(addr)) {
+			members[addr].readyToSyncTimestamp = now;
+			return _rankValidator(addr);
```

Comment:
consider adding here:
`members[addr].readyForCommittee = false`
as a mean to for a standby to notify going out of sync.
If we do this, standbys which are ready-for-committe should not send ready-to-sync as a heartbeat transaction, but ready-for-committee instead.